lib/workers/repository/updates/generate.ts

```diff
@@ -115,7 +115,7 @@ export function generateBranchConfig(
         upgrade.newDigestShort ||
         upgrade.newDigest.replace('sha256:', '').substring(0, 7);
     }
-    if (upgrade.isDigest) {
+    if (upgrade.updateType === 'pinDigest' && !upgrade.currentDigest) {
```

Comment:
```suggestion
    if (upgrade.updateType === 'pinDigest') {
```

Reply:
But what about digest updates? They still exist and are more common. I don't see how you can remove the "if upgrade.isDigest" logic without it changing behavior for digest->digest updates

Reply:
I see I need to check this case.
I thought this if condition was causing the bug but I think now that we should do an or condition here for the pin digests case ,cause when it was set we were pinning tags.

Reply:
```suggestion
    if (upgrade.isDigest || upgrade.updateType === 'pinDigest') {
```
The tests that I've edited were meant for when the `pinDigests` config flag was set to true.
In the case when `pinDigests` was set to true the `upgrade.isDigest` in that if was undefined and the values for the change column were being set to the tag.
therefore I added new update type `pinDigest` which I've added here in case where where pinDigest is set : 
https://github.com/renovatebot/renovate/blob/426dcd20be2a47c9d083de4158da910b98753fd2/lib/workers/repository/process/lookup/index.ts#L337
That sound right ?

Reply:
Yes, I think that adding updateType=pinDigest is logically correct. But you need to make sure you retain backwards compatibility:
- pin and digest updateTypes should behave exactly as before
- pinDigest and pin updateTypes should remain together in a "Pin dependencies" PR

Reply:
@rarkins Would you please elaborate more on the second bullet that you've mentioned ?
I thought this was only a cosmetic fix. In the case where `pinDigests` was set we were not setting the `upgrade.displayFrom` and `upgrade.displayTo` to the digests, that wasn't the issue ?.

Reply:
Today, both version pins and digest pins are grouped together into the same PR. Your changes need to maintain that backwards compatibility. You could check that by having a repo with one pinned version (e.g. npm `^1.0.0` -> `1.0.1`) and one pinned digest and verifying that they are grouped together like before